source/defclass-star.lisp

```diff
@@ -33,10 +33,20 @@ Use the slot name directly:
 (defvar *accessor-name-transformer* 'default-accessor-name-transformer)
 (defvar *automatic-accessors-p* #t)
 
+(defvar *predicate-name-transformer* 'default-predicate-name-transformer
+  "A function that takes the class name and its definition as argument.
+Return the predicate name.
+The predicate returns non-nil when the argument is of the `name' class.")
+
+(defun default-predicate-name-transformer (name &rest args)
+  (declare (ignore args))
+  (intern (format nil "~a-P" name) *package*))
```

Comment:
@Ambrevar did you think through whether there is any guarantee about the value of *package* at all macroexpand times? (SBCL has a tendency to call macroexpand at multiple times)

i'll merge this, but my gut feeling is telling me that it's a potential source of hard-to-debug bugs where the predicate ends up interned into a surprise package, and it may even redefine something there... reminds me of the longest wasted debug sessions in CL, and what i learned from them: if something doesn't make any sense, then always assert the symbol packages first... :)

Reply:
You are right, it should be `(symbol-package name)` instead of `*package*`, so that the predicate belongs to the same package as `name`.  I'll send a fix.